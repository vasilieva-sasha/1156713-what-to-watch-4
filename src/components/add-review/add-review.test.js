import React from "react";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import configureStore from 'redux-mock-store';
import NameSpace from './../../reducer/name-space';
import AddReview from './add-review';

const film = {
  title: ``,
  posterInfo: ``,
  background: ``,
  backgroundColor: ``
};

const mockStore = configureStore([]);

describe(`AddReviewComponent`, () => {
  const store = mockStore({
    [NameSpace.APP]: {
      currentPage: `REVIEW`,
      isFormBlocked: false
    },
    [NameSpace.DATA]: {
      serverError: false
    },
    [NameSpace.USER]: {
      authorizationStatus: `AUTH`,
      authInfo: {
        avatar: ``
      }
    }
  });
  it(`Add review renders correctly`, () => {
    const tree = renderer.create(
        <Provider store={store}>
          <AddReview
            film={film}
            onSubmit={() => {}}
            onRatingSelect={() => {}}
            onCommentType={() => {}}
            isButtonBlocked={false}
            isFormBlocked={store.isFormBlocked}
            currentPage={store.currentPage}
            serverError={store.serverError} />
        </Provider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
