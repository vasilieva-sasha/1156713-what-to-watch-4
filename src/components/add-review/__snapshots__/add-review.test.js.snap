// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddReviewComponent Add review renders correctly 1`] = `
<section
  className="movie-card movie-card--full"
  style={
    Object {
      "background": "",
    }
  }
>
  <div
    className="movie-card__header"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt=""
        src=""
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header movie-card__head"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
          href="main.html"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <nav
        className="breadcrumbs"
      >
        <ul
          className="breadcrumbs__list"
        >
          <li
            className="breadcrumbs__item"
          >
            <a
              className="breadcrumbs__link"
              href="movie-page.html"
            >
              
            </a>
          </li>
          <li
            className="breadcrumbs__item"
          >
            <a
              className="breadcrumbs__link"
            >
              Add review
            </a>
          </li>
        </ul>
      </nav>
      <div
        className="user-block"
      >
        <div
          className="user-block__avatar"
        >
          <img
            alt="User avatar"
            height="63"
            src=""
            width="63"
          />
        </div>
      </div>
    </header>
    <div
      className="movie-card__poster movie-card__poster--small"
    >
      <img
        alt=" poster"
        height="327"
        src=""
        width="218"
      />
    </div>
  </div>
  <div
    className="add-review"
  >
    <form
      action="#"
      className="add-review__form"
      disabled={false}
      onSubmit={[Function]}
    >
      <div
        className="rating"
      >
        <div
          className="rating__stars"
          onChange={[Function]}
        >
          <input
            className="rating__input"
            id="star-1"
            name="rating"
            type="radio"
            value="1"
          />
          <label
            className="rating__label"
            htmlFor="star-1"
          >
            Rating 1
          </label>
          <input
            className="rating__input"
            id="star-2"
            name="rating"
            type="radio"
            value="2"
          />
          <label
            className="rating__label"
            htmlFor="star-2"
          >
            Rating 2
          </label>
          <input
            className="rating__input"
            id="star-3"
            name="rating"
            type="radio"
            value="3"
          />
          <label
            className="rating__label"
            htmlFor="star-3"
          >
            Rating 3
          </label>
          <input
            className="rating__input"
            id="star-4"
            name="rating"
            type="radio"
            value="4"
          />
          <label
            className="rating__label"
            htmlFor="star-4"
          >
            Rating 4
          </label>
          <input
            className="rating__input"
            id="star-5"
            name="rating"
            type="radio"
            value="5"
          />
          <label
            className="rating__label"
            htmlFor="star-5"
          >
            Rating 5
          </label>
        </div>
      </div>
      <div
        className="add-review__text"
        style={
          Object {
            "background": "rgba(255, 255, 255, 0.1)",
          }
        }
      >
        <textarea
          className="add-review__textarea"
          id="review-text"
          maxLength="400"
          minLength="50"
          name="review-text"
          onChange={[Function]}
          placeholder="Review text"
        />
        <div
          className="add-review__submit"
        >
          <button
            className="add-review__btn"
            disabled={false}
            type="submit"
          >
            Post
          </button>
        </div>
      </div>
    </form>
    
  </div>
</section>
`;
